lib.php

```diff
@@ -0,0 +1,61 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Libs.
+ * @copyright  Catalyst IT 2022
+ * @package    local_login
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+/**
+ * Triggers after site config is loaded.
+ *
+ */
+function local_login_after_config() {
+    global $CFG, $SESSION, $_SERVER;
+    $errorcode = 0;
+    $noredirect  = optional_param('noredirect', 0, PARAM_BOOL); // don't redirect
+
+    if (!isloggedin() && empty($noredirect)) {
+        // Set new login URL.
+        $loginurl = new moodle_url('/local/login/index.php');
+        $loginurlstr = $loginurl->out(false);
+
+        // To stop redirect if alternateloginurl is set.
+        if (!empty($CFG->alternateloginurl)) {
+            unset($CFG->alternateloginurl);
+        }
+
+        if (!empty($SESSION->wantsurl) && strpos($SESSION->wantsurl, $loginurlstr) === 0) {
+            // To stop the redirect loop as local/login is redirected from login page and this causes an unending redirect.
+            $SESSION->wantsurl = null;
+        } else if (isset($_SERVER['REQUEST_URI'])) {
+            // To avoid this part being called from other plugins like workshopform_rubric_testsuite.
```

Comment:
when the check is only triggered when noredirect = 1 (like I suggest above) I don't think you will need this as it won't be triggered by other scripts?